config/examples/Creality/CR-10 V3/Configuration.h

```diff
@@ -1187,17 +1187,17 @@
  *     |    [-]    |
  *     O-- FRONT --+
  */
-#define NOZZLE_TO_PROBE_OFFSET { 47, 0, 0 }
+#define NOZZLE_TO_PROBE_OFFSET { 46, 2, 0 }
 
 // Most probes should stay away from the edges of the bed, but
 // with NOZZLE_AS_PROBE this can be negative for a wider probing area.
-#define PROBING_MARGIN 10
+#define PROBING_MARGIN 45
```

Comment:
Adjust for the more extended probing area possible with cr-10 v3

Reply:
this change would shrink the probing area by 90mm not extend it, probing margin moves the probing area in from all edges of the bed this many mm. this feature is largely over used when the build volume should have been directly changed allowing for the proper area to be probed. generally the only probe that would require the use of margin is any probe that does not physically touch the bed as they do not read near the edge.

Reply:
I have the machine next to me I can confirm my probing  area is not shrinked by 90mm.
This because of nozzle and probe offset. 
If i'm hitting the endstop X, the probe is already +/- 40mm in the print area

Reply:
probe margin only effects mesh bed leveling probing area, i do not believe you fully understand a number of the changes made.

Reply:
If the probe is already 40mm IN the print area when the X carriage is pushing the endstop, how could you probe less than 40mm ?

Reply:
i don't think you are understanding the relation of probe offsets and probe_margin and you are referencing things that are unrelated.

Reply:
But If I put 40, then I'm hitting the endstop during the leveling. How do you explain that ?

Reply:
its a combinations of other suspect settings that result in a behavior you consider abnormal. 
i have 4 printers with a 40mm probe offsets and have probe_margin 100% disabled not even set to 0.
i am hold a firm stance that both margin and inset should never be used with a probe that physically touches the bed a proper well define build volume is the solution.
the over use of both margin and inset for less then intended reasons has lead to many other issues that in turn get reported a bugs that are really user error.

Reply:
Actually both will work. But with 10mm margin only Marlin has to recalculate the probing area.

If you have a 300x300 defined print surface and a 10mm probing margin, Marlin will want to probe a 280X280 square. One probing every 70mm (with 5x5 points probing).
As the endstops are active during bed leveling, Marlin will realize that it's not possible to reach X10 position (with 47mm probe X offset). He will then recalculate the probing area to go from the minimum physically possible probing position to the the set margin on the right of the print area. 
In the case of the CR-10 V3 that will give one probing every +/- 63mm measure when doing a real test with 10mm margin.

Marlin does : 
- Ok I'm trying to probe at X10 (with probe offset, So X-36 from nozzle point of view) but I'm reaching X Endstop before, so it means right now I'm at X-5 as defined with MIN_POS_X (nozzle position)
- I have a 46mm x probe Offset
- I means the min X position I can reach with the probe is : -5+46 = 41mm
- So my mesh will have to be from X41 to X290 (10mm margin)
- Divided by 4 (for five point) the result  is 62,25, strangely almost the same than what have been measured in real test.

With 45mm margin, it's giving a probe area about 210*210. Divided by 4 it's giving a theoric 60mm between probe points and it is what the machine exactly does when you try and measure. 

So with 10mm Margin : 
- The probing area is not centered to the print area.
- The center of the probing area is not aligned with the HOME probing position.
- The probing area is not square anymore
- The first Y probing line of the mesh will be 41mm from the edge of the print area when the last one will 10mm close to the edge.
- You are decreasing the probing density on the most used printing area. (It's not all days that you're printing pieces close to the print area dimensions).

You can see also that if you try to compile with the tramming wizard margin at 10 mm, like this : 
#define TRAMMING_POINT_XY { { 10, 10 }, { 290, 10 }, { 290, 290 }, { 10, 290 } }

You will get an error saying that those positions are not compatible with the defined probe offset : 

Marlin\src\gcode\bedlevel\../../feature/tramming.h:36:36: error: static assertion failed: TRAMMING_POINT_XY point 0 is not reachable with the default NOZZLE_TO_PROBE offset and PROBING_MARGIN.

So I'm still convinced those 45mm margin are correct and suitable to get something clean and centered.

Reply:
you seem to be under the impression several settings work in a way that they do not.
furthermore the use of both corner leveling and the tramming wizard is semi redundant they are 2 features that do the same thing just with a different method.
given that marlin must have at least 1kb of free ram for proper functionality the addition of extraneous features is not advised.
GitHub really isn't the place for this type of discussion i would advise joining the marlin discord and discussing all of this with the devs & documentarians, that include my self. 
you seem to be under the impression your being helpful but due to your lack of experience and complete understanding, i need to inform you a number of these changes will ultimately lead to many bug reports.

Reply:
With a +46mm probe X offset, the probe cannot reach the left side but it can definitely reach the right side, so you can set a very small margin, and only the left margin needs to be large. Give this a try:
```cpp
#define PROBING_MARGIN 10
#define PROBING_MARGIN_LEFT 45
```